service/service.go

```diff
@@ -313,19 +338,36 @@ func (app *Application) notifyPipelineNotReady() {
 	}
 }
 
-func (app *Application) shutdownPipelines() {
+func (app *Application) shutdownPipelines() error {
 	// Shutdown order is the reverse of building: first receivers, then flushing pipelines
 	// giving senders a chance to send all their data. This may take time, the allowed
 	// time should be part of configuration.
 
+	var errs []error
+
 	app.logger.Info("Stopping receivers...")
-	app.builtReceivers.StopAll()
+	err := app.builtReceivers.StopAll()
+	if err != nil {
+		errs = append(errs, errors.Wrap(err, "failed to stop receivers"))
+	}
 
 	app.logger.Info("Stopping processors...")
-	app.builtPipelines.ShutdownProcessors(app.logger)
+	err = app.builtPipelines.ShutdownProcessors(app.logger)
+	if err != nil {
+		errs = append(errs, errors.Wrap(err, "failed to shutdown processors"))
+	}
 
 	app.logger.Info("Shutting down exporters...")
-	app.exporters.ShutdownAll()
+	err = app.exporters.ShutdownAll()
+	if err != nil {
+		errs = append(errs, errors.Wrap(err, "failed to shutdown exporters"))
+	}
+
+	if len(errs) != 0 {
+		return oterr.CombineErrors(errs)
+	}
+
+	return nil
 }
 
 func (app *Application) shutdownExtensions() {
```

Comment:
Should this function also return errors rather than `app.logger.Warn(...)`?
Or switch other shutdown functions to logging as well for consistency?

Reply:
For consistency, it should return errors too.

Reply:
Done.